src/PowerShellEditorServices/Language/FindFoldsVisitor.cs

```diff
@@ -0,0 +1,121 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+
+using System;
+
+namespace Microsoft.PowerShell.EditorServices
+{
+    /// <summary>
+    /// The visitor used to find the all folding regions in an AST
+    /// </summary>
+    internal class FindFoldsVisitor : AstVisitor
+    {
+        private const string RegionKindNone = null;
+
+        public List<FoldingReference> FoldableRegions { get; private set; }
```

Comment:
Probably no need for the `private set` here if the only assignment is in the constructor. Don't need a setter for that.

Reply:
Fixed